src/main/java/me/zeroeightsix/kami/util/FakeMapItemRenderer.java

```diff
@@ -0,0 +1,187 @@
+/*
+Taken from net.minecraft.client.gui.MapItemRenderer
+ */
+
+package me.zeroeightsix.kami.util;
+
+import com.google.common.collect.Maps;
+import me.zeroeightsix.kami.module.modules.render.NoMaps;
+import net.minecraft.block.material.MapColor;
+import net.minecraft.client.gui.MapItemRenderer;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.texture.DynamicTexture;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.MapData;
+import net.minecraft.world.storage.MapDecoration;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nullable;
+import java.util.Iterator;
+import java.util.Map;
+
+import static me.zeroeightsix.kami.KamiMod.MODULE_MANAGER;
+
+public class FakeMapItemRenderer extends MapItemRenderer {
```

Comment:
I'll clean this up myself when I'm on a computer

Reply:
I am just going to point out that you cannot inject into this, at least, I could not figure it out. By default, `MapItemRenderer.Instance` is a private class in addition to being nested and therefore cannot be mixined. I just changed that so if someone else ever wants to make changes to the Minecraft `MapItemRenderer` then they can just mixin the FakeMapItemRenderer. I'll let you attempt it though, I am sure you are more skilled than I am.

Reply:
I agree, it is half the fun lol. You can change it to public with an access transformer, but I've never tested if those work with mixins

Reply:
I read somewhere that you can, I know nothing about access transformers though; I may want to look them up later.